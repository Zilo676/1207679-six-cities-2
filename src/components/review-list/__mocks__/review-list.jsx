import React from "react";

const ReviewList = () => {
  return <></>;
};

export {ReviewList};
export default ReviewList;
