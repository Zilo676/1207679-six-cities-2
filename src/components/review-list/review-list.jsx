import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {Review} from '../review/review.jsx';
import {getComments} from '../../reducer/comments/selectors';

const ReviewList = (props) => {
  const {comments} = props;

  return (
    comments.map((review, i) => (<Review key={i} review={review} />))
  );
};

ReviewList.propTypes = {
  comments: PropTypes.array,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  comments: getComments(state),
});

export {ReviewList};

export default connect(mapStateToProps, null)(ReviewList);
