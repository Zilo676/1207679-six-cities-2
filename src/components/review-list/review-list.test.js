import React from 'react';
import renderer from 'react-test-renderer';

import {ReviewList} from './review-list.jsx';

jest.mock(`../review/review.jsx`, () => () => `Review`);

it(`Correctly rendered`, () => {
  const tree = renderer.create(<ReviewList
    comments={[]}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
