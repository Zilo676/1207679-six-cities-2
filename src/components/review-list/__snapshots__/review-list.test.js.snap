// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered 1`] = `
<h2
  className="reviews__title"
>
  Reviews Â· 
  <span
    className="reviews__amount"
  >
    0
  </span>
</h2>
`;
